include/d_profileNP/bases/d_yes_no_window.hpp

```diff
@@ -0,0 +1,161 @@
+#include <dol/bases/d_lytbase.hpp>
+#include <dol/sLib/s_State.hpp>
+
+/// @brief Displays a yes/no window.
+/// @details This appears, for example, in the pause menu or when you press + on the world map.
+/// @ingroup bases
+class dYesNoWindow_c : public dBase_c {
+public:
+    /// @brief The type of the window.
+    enum WINDOW_TYPE_e {
+        SAVE_DATA_CREATED,
+        SAVE,
+        SAVED,
+        EXIT_FREE_MODE,
+        WATCH_UNLOCKED_HINT_MOVIE,
+        SKIP_COURSE,
+        WATCH_HINT_MOVIE,
+        GOT_ALL_STAR_COINS,
+        GOT_ALL_STAR_COINS_MENTION_W9,
+        BOWSERJR_SINGLEPLAYER,
+        RETURN_TITLE,
+        BOWSERJR_MULTIPLAYER,
+        RETURN_TO_MAP,
+        BACK_TO_COURSE_SELECTION,
+        GOT_ALL_STAR_COINS_ALL_WORLDS,
+        QUICK_SAVE,
+        QUICK_SAVED,
+        ALL_TOAD_HOUSES_RETURNED,
+        ITEM_INFO,
+        COURSE_CLEAR_WITH_SUPER_GUIDE,
+        SKIP_COURSE_CONFIRM,
+        COMPLETED_EVERYTHING,
+        PROGRESS_AUTOSAVE_AFTER_W9_UNLOCK,
+        BLOCK_TRANSFORMATION,
+        BLOCK_TRANSFORMATION_REVERSE,
+        EXIT_COIN_BATTLE,
+        RETURN_TO_MAP2,
+        PEACH_CASTLE_HINT1,
+        PEACH_CASTLE_HINT2,
+        NUM_WINDOW_TYPES
+    };
+    
+    /// @brief The possible cursor positions.
+    enum CURSOR_POS_e {
+        POS_YES,
+        POS_NO,
+        POS_OK
+    };
+
+    /// @brief The sound effects used for the yes/no window.
+    enum SOUND_e {
+        SOUND_CLOSE,
+        SOUND_WAIT,
+        SOUND_CURSOR_MOVE,
+        SOUND_OPEN_ANIME
+    };
+
+    /// @brief The animations used for the yes/no window.
+    enum ANIMS_e {
+        ANIM_IN_WINDOW,
+        ANIM_IN_QUICK_SAVE,
+        ANIM_LOOP_WINDOW,
+        ANIM_LOOP_QUICK_SAVE,
+        ANIM_IN_YES,
+        ANIM_IN_NO,
+        ANIM_IN_OK,
+        ANIM_ON_YES,
+        ANIM_ON_NO,
+        ANIM_ON_OK,
+        ANIM_IDLE_YES,
+        ANIM_IDLE_NO,
+        ANIM_IDLE_OK,
+        ANIM_HIT_YES,
+        ANIM_HIT_NO,
+        ANIM_HIT_OK,
+        ANIM_HIT_QUICK_SAVE,
+        ANIM_OFF_YES,
+        ANIM_OFF_NO,
+        ANIM_OFF_OK,
+        ANIM_IN_BG,
+        ANIM_OUT_BG,
+        ANIM_OUT_WINDOW
+    };
+
+    dYesNoWindow_c();
+    virtual ~dYesNoWindow_c();
+    virtual int create();
+    virtual int execute();
+    virtual int draw();
+    virtual int doDelete();
+
+private:
+    /// @brief Fills in the layout of the window.
+    /// @details This includes:
+    /// - Filling out the text boxes
+    /// - Setting the visibility of the buttons
+    /// - Conditionally showing the quick save balloon
+    void populateLayout();
+
+    STATE_FUNC_DECLARE(dYesNoWindow_c, InitWait); ///< @brief Initial state. Switches to OpenAnimeStartWait immediately.
+    STATE_FUNC_DECLARE(dYesNoWindow_c, OpenAnimeEndWait); ///< @brief Waiting for the window to finish the opening animation.
+    STATE_FUNC_DECLARE(dYesNoWindow_c, ButtonOnStageAnimeEndWait); ///< @brief Playing the button to finish the appearing animation.
```

Comment:
```suggestion
    STATE_FUNC_DECLARE(dYesNoWindow_c, ButtonOnStageAnimeEndWait); ///< @brief Waiting for the button to finish the appearing animation.
```
That's my assumption of what this means at least :P